### DISCUSSION TITLE
Returning JSX from meta and link export functions

### DISCUSSION THREAD

**Discussion by subhero24** [remix-run/remix#2470] [Proposals]
Currently meta and link export functions return an array with objects that describe the meta and link tags for the head of the document:

```javascript
export function links() {
  return [
    {
      rel: "stylesheet",
      href: "style.css",
    },
    {
      rel: "stylesheet",
      href: "tailwind.css",
    },
  ];
}
```

Wouldn't it be nicer if these function would return the jsx instead of the javascript objects for these elements?

```javascript
export function links() {
  return <>
    <link rel="stylesheet" href="style.css" />
    <link rel="stylesheet" href="tailwind.css" />
  </>
}
```

**Comment by sergiodxa** [comment]
If LinksFunction, or MetaFunction, let you return JSX, then it would be no way for Remix to ensure you only render links or meta tags there and not other things.

Right now, by using an array of objects, it can check that those objects are correct and only put links or meta tags there.

Basically, you could render a whole app as a link by allowing JSX.

**Reply by subhero24** [reply]
Remix could just inspect the returned element to make sure only link or head elements are returned.
However, i do not think the overhead of checking this is worth it. Why not just make this a convention, like all the other conventions that come with a framework? I think it would be a much more readable than the object representation.
